### PR TITLE
Threading corrections for GM and Redi

### PR DESCRIPTION
**Pull request by mark-petersen**
Two loops in GM and Redi did not have the correct OpenMP pragmas.

### CODE REVIEW

**Comment by mark-petersen** [comment]
@jonbob or @vanroekel, this was not causing the current problem, but I wanted to correct it now anyway. Please take a quick look. I think these were the only MPAS corrections to https://github.com/E3SM-Project/E3SM/pull/3651, unless you have seen others.

File: src/core_ocean/shared/mpas_ocn_gm.F [hunk 1]
```diff
@@ -490,7 +490,8 @@ subroutine ocn_GM_compute_Bolus_velocity(statePool, diagnosticsPool, &
            call mpas_pool_get_array(diagnosticsPool, 'RiTopOfCell', RiTopOfCell)
            call mpas_pool_get_array(diagnosticsPool, 'zTop', zTop)
            call mpas_pool_get_array(diagnosticsPool, 'c_visbeck', c_visbeck)
-           !$omp do schedule(runtime) private(cell1, cell2, sumN2, ltSum, sumRi, countN2, zEdge, RiTopOfEdge, BruntVaisalaFreqTopEdge) 
+           !$omp do schedule(runtime) private(k, cell1, cell2, sumN2, ltSum, sumRi, countN2, &
+           !$omp& zEdge, RiTopOfEdge, BruntVaisalaFreqTopEdge)
```
Comment (jonbob) [inline src/core_ocean/shared/mpas_ocn_gm.F:494]: @mark-petersen - I think you need to add k to the omp pragmas at 424 and 452 above as well?

File: src/core_ocean/shared/mpas_ocn_tracer_hmix_redi.F [hunk 2]
```diff
@@ -180,6 +180,7 @@ subroutine ocn_tracer_hmix_Redi_tend(meshPool, scratchPool, layerThickness, laye
       !$omp barrier
 
       nCells = nCellsArray(2)
+      !$omp do schedule(runtime) private(i, iEdge)
```
Comment (jonbob) [inline src/core_ocean/shared/mpas_ocn_tracer_hmix_redi.F:183]: @mark-petersen - I'm not sure I understand the omp master section above, why some allocates are inside it and others are not

**Comment by mark-petersen** [comment]
@jonbob thanks for the comments. I'm out of time today, but we can continue tomorrow with a fix-up.

**Comment by mark-petersen** [comment]
This was included in #620.

**Pull request closed without merging**